src/common/k8s-api/kube-object.store.ts

```diff
@@ -138,41 +157,41 @@ export abstract class KubeObjectStore<T extends KubeObject> extends ItemStore<T>
     return [];
   }
 
-  getById(id: string) {
+  getById(id: string | undefined): K | undefined {
```

Comment:
Seems weird that `getById(undefined)` works?

Reply:
Fair enough. This is probably a "wrong level of abstraction" problem. There are a lot of places that we passed in a value which we got directly from an object where the field is optional (and then maybe undefined).

I chose this direction as the path of least resistance.